src/extra/components/_adq.pyx

```diff
@@ -0,0 +1,42 @@
+
+# cython: boundscheck=False, wraparound=False, cdivision=True, initializedcheck=False
+
+from cython cimport numeric
+from libc.limits cimport INT_MAX
+from libc.string cimport memcpy
+
+
+def _reshape_flat_pulses(
```

Comment:
Unfortunately I no longer find my documentation on this, but I kept this Cython implementation around as it proved significantly faster than a numpy implementation. Note that this method supports a varying number of pulses given the flat shape, while the other reshaping style does not.